backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -1132,9 +1126,13 @@ export class SegmentService {
     // Check if the segment name already exists in the same context
     // If id is present, check only for other segments with the same name and context
     if (id) {
-      sameNameSegment = await this.segmentRepository.find({ where: { name, context, id: Not(id) } });
+      sameNameSegment = await this.segmentRepository.find({
+        where: { name, context, type: Not(SEGMENT_TYPE.PRIVATE), id: Not(id) },
```

Comment:
ok, this is because private segments can be named same thing right? makes sense.

Reply:
yep, and we don't to be able to create a regular segment with the same name as a global exclude one